Userland/Services/WebContent/RenderLoopExecutor.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2024, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibCore/Forward.h>
+#include <LibThreading/ConditionVariable.h>
+#include <LibThreading/MutexProtected.h>
+#include <LibThreading/Thread.h>
+#include <LibWeb/Painting/RecordingPainter.h>
+#include <WebContent/Forward.h>
+
+#ifdef HAS_ACCELERATED_GRAPHICS
+#    include <LibAccelGfx/Context.h>
+#endif
+
+namespace WebContent {
+
+class RenderLoopExecutor {
+    AK_MAKE_NONCOPYABLE(RenderLoopExecutor);
+    AK_MAKE_NONMOVABLE(RenderLoopExecutor);
+
+public:
+    struct BackingStores {
+        i32 front_bitmap_id { -1 };
+        i32 back_bitmap_id { -1 };
+        RefPtr<Gfx::Bitmap> front_bitmap;
+        RefPtr<Gfx::Bitmap> back_bitmap;
+    };
+
+    void paint(NonnullOwnPtr<Web::Painting::RecordingPainter> recording_painter, Function<void(i32)> on_completion);
```

Comment:
This feels like there's an opportunity to use Core::Promise and resolve it with the bitmap id rather than passing an explicit completion handler function.